src/service/viewport-impl.js

```diff
@@ -134,6 +134,10 @@ export class Viewport {
 
     this.binding_.onScroll(this.scroll_.bind(this));
     this.binding_.onResize(this.resize_.bind(this));
+
+    this.onScroll(() => {
+      this.viewer_.postScroll(this.getScrollTop());
```

Comment:
Animation frames are "pumped" by the browser when it wants to actually draw something to the screen. You can get one via `requestAnimationFrame` and our `vsync` library wraps it.

You want an algorithm where we never send a scroll event when we already did so in the current frame (but you need to schedule a next frame to send the last received event if you dropped a scroll event from sending.